ethsigner/core/src/integration-test/java/tech/pegasys/ethsigner/jsonrpcproxy/IntegrationTestBase.java

```diff
@@ -335,4 +343,34 @@ private static File createFile(final String s) throws IOException {
     file.deleteOnExit();
     return file;
   }
+
+  private static int httpJsonRpcPort() {
+    final File portsFile = new File(dataPath.toFile(), PORTS_FILENAME);
+    LOG.info("Awaiting presence of ethsigner.ports file: {}", portsFile.getAbsolutePath());
+    awaitPortsFile(dataPath);
+    LOG.info("Found ethsigner.ports file: {}", portsFile.getAbsolutePath());
+
+    try (final FileInputStream fis = new FileInputStream(portsFile)) {
+      final Properties portProperties = new Properties();
+      portProperties.load(fis);
+      final String value = portProperties.getProperty(HTTP_JSON_RPC_KEY);
+      return Integer.parseInt(value);
+    } catch (final IOException e) {
+      throw new RuntimeException("Error reading Web3Provider ports file", e);
+    }
+  }
+
+  private static void awaitPortsFile(final Path dataDir) {
```

Comment:
_technicaly_ this is "waitForFileToExist"

Reply:
renamed it and now take a path as a arg